lanternmq/rabbitmq/messagequeue.go

```diff
@@ -0,0 +1,346 @@
+	if idInt >= len(mq.channels) {
+		return nil, errors.New("no channel with the requested ID was found")
+	}
+	ch := mq.channels[idInt]
+	return ch, nil
+}
+
+// Connect creates a connection to a RabbitMQ service.
+func (mq *MessageQueue) Connect(username string, password string, host string, port string) error {
+	s := fmt.Sprintf("amqp://%s:%s@%s:%s/", username, password, host, port)
+	conn, err := amqp.Dial(s)
+	if err != nil {
+		err = errors.New("unable to connect to message queue")
+	}
+	mq.connection = conn
+
+	return err
+}
+
+// CreateChannel creates a channel to the RabbitMQ service that has already been connected to.
+// If the RabbitMQ service has not been connected to already, an error is thrown.
+// The channel's ID is returned.
+func (mq *MessageQueue) CreateChannel() (lanternmq.ChannelID, error) {
+	var err error
+	if mq.connection == nil {
+		err = errors.New("connection must exist before creating a channel")
+		return "", err
+	}
+	ch, err := mq.connection.Channel()
+	if err != nil {
+		err = errors.New("unable to create channel")
+		return "", err
+	}
+	id, err := mq.addChannel(ch)
+
+	return id, err
+}
+
+// NumConcurrentMsgs defines how many messages the user can process from the channel at one time.
+func (mq *MessageQueue) NumConcurrentMsgs(chID lanternmq.ChannelID, num int) error {
+	ch, err := mq.getChannel(chID)
+	if err != nil {
+		return err
+	}
+
+	err = ch.Qos(
+		num,   // prefetch count
+		0,     // prefetch size
+		false, // global
+	)
+	if err != nil {
+		err = errors.New("unable to set the number of concurrent messages that can be handled")
+		return err
+	}
+	return err
+}
+
+// DeclareQueue creates a queue with the given name on the given channel using RabbitMQ's
+// QueueDeclare method with the following arguments:
+// * name: qName
+// * durable: true
+// * autoDelete: false
+// * exclusive: false
+// * noWait: false
+// * args: nil
+func (mq *MessageQueue) DeclareQueue(chID lanternmq.ChannelID, qName string) error {
+	ch, err := mq.getChannel(chID)
+	if err != nil {
+		return err
+	}
+
+	_, err = ch.QueueDeclare(
+		qName, // name
+		true,  // durable
+		false, // delete when unused
+		false, // exclusive
+		false, // no-wait
+		nil,   // arguments
+	)
+	if err != nil {
+		err = fmt.Errorf("unable to create queue: %s", err.Error())
+		return err
+	}
+	return err
+}
+
+// PublishToQueue publishes 'message' on the queue with name 'qName' over the channenl with ID 'chID'
+// by calling the RabbitMQ Publish method with the following arguments:
+// exchange: ""
+// key: qName
+// mandatory: false
+// immediate: false
+// publishing:
+//   DeliveryMode: amqp.Persistent
+//   ContentType: "text/plain"
+//   Body: []byte(message)
+func (mq *MessageQueue) PublishToQueue(chID lanternmq.ChannelID, qName string, message string) error {
+	ch, err := mq.getChannel(chID)
+	if err != nil {
+		return err
+	}
+
+	err = ch.Publish(
+		"",    // exchange
+		qName, // routing key
+		false, // mandatory
+		false, // immediate
```

Comment:
If you're not going to support the caller changing these options, use declared constants instead of straight Booleans.